core/src/main/java/de/danoeh/antennapod/core/feed/FeedMedia.java

```diff
@@ -579,4 +580,12 @@ private void checkEmbeddedPicture() {
             hasEmbeddedPicture = Boolean.FALSE;
         }
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof RemoteMedia) {
```

Comment:
I didn't care about changing the behavior for the current classes, so all I did in this case was to direct to the same method on RemoteMedia that compares instances of these two objects. All it really compares is the feed link and the episode identifier, I believe.